sno/working_copy/gpkg_adapter.py

```diff
@@ -474,29 +485,111 @@ def json_to_gpkg_metadata(v2_metadata_json, table_name, reference=False):
     return result
 
 
+_KNOWN_METADATA_URIS = {
+    "GDALMultiDomainMetadata": "http://gdal.org",
+}
+
+
+def _get_uri_from_xml(xml_metadata):
+    from xml.dom.minidom import parseString
+
+    try:
+        doc = parseString(xml_metadata)
+        element = doc.documentElement
+        return (
+            _KNOWN_METADATA_URIS.get(element.tagName)
+            or element.getAttribute("xmlns")
+            or element.namespaceURI
+            or "(unknown)"
+        )
+    except Exception:
+        return "(unknown)"
+
+
+def xml_to_gpkg_metadata(metadata_xml, table_name, reference=False):
+    if reference:
+        timestamp = datetime_to_iso8601_utc(datetime.now())
```

Comment:
Feels like this should probably be the commit timestamp of the last change to the metadata blob?

Reply:
Not yet done - this is a bigger change than you would think since it means the working-copy generated is no longer a function of a particular git tree, but of the entire repo history - need to wire through some more data to the right places.

Will do this in a separate PR.

Reply:
Is it worth doing?